### PR TITLE
Add Sherlock provider for locator suggest endpoint

### PR DESCRIPTION
**Pull request by stdavis**
This was tested using masquerade but it should work with any esri locator service.

### CODE REVIEW

**Comment by stdavis** [comment]
Actually, just found a bug...

**Comment by stdavis** [comment]
OK. This is ready for another look.

**Comment by stdavis** [comment]
@steveoh This is ready for a look when you get some time.

File: packages/sherlock/src/providers.js [hunk 1]
```diff
@@ -258,15 +258,23 @@ class LocatorSuggestProvider extends (ProviderBase) {
...
+          attributes: suggestion
+        };
+      });
+
+      return { ok: true, data: features };
+    } catch {
+      const message = 'error with suggest request';
+      console.error(message , response);
+
+      throw new Error(message);
```
Comment (steveoh) [inline packages/sherlock/src/providers.js:276]: **question**: should this return `ok: false` with the error message? or does the main code expect the error to be thrown? It seems like we have two competing patterns here

File: packages/sherlock/src/Sherlock.js [hunk 2]
```diff
@@ -251,7 +253,7 @@ class Clue extends Component {
       short: clue.length <= 2,
       hasmore: hasMore
     });
-  });
+  }, 100);
```
[the remark this answers is not in the document]
Comment (stdavis) [reply packages/sherlock/src/Sherlock.js:256]: This is the second argument to `debounce`, not `setState`. Just trying to cut down on unneeded requests if the user is actively typing.

File: packages/sherlock/src/providers.js (hunk 1 above)
Comment (stdavis) [reply packages/sherlock/src/providers.js:276]: The main code doesn't even check the `ok` prop. It expects an error to be thrown. Yes, I noticed that the `ok` stuff wasn't consistent. I can take another look and maybe just remove it and make sure that errors are thrown.

File: packages/sherlock/src/Sherlock.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (steveoh) [reply packages/sherlock/src/Sherlock.js:256]: 😆

File: packages/sherlock/src/providers.js (hunk 1 above)
Comment (stdavis) [reply packages/sherlock/src/providers.js:267]: OK. I gave it a try. Let me know what you think.

File: package.json [hunk 3]
```diff
@@ -10,29 +10,32 @@
...
+    "copy-webpack-plugin": "^6.*",
     "copyfiles": "^2.4.1",
     "core-js": "^3.8.2",
     "eslint": "^7.17.0",
     "esri-loader": "^3.0.0",
     "identity-obj-proxy": "^3.0.0",
     "jest": "^26.6.3",
     "lerna": "^3.22.1",
+    "react": "*",
+    "react-dom": "^17.0.2",
```
Comment (steveoh) [inline package.json:38]: **question**: will you explain this one to me

Comment (stdavis) [reply package.json:38]: I was having troubles getting storybook to run. Let me take a look and see if this is still needed...

Comment (stdavis) [reply package.json]: I removed these...

**Review by steveoh — APPROVED** [review]

**Pull request merged**